rust/src/host_functions.rs

```diff
@@ -61,5 +64,10 @@ pub mod host_functions_impl {
             buf.copy_from_slice(&digest);
             buf
         }
+
+        fn blake3hash(message: &[u8]) -> [u8; 32] {
+            let hash = blake3::hash(message);
+            *hash.as_bytes()
```

Comment:
Should this be done (for consistency) like for the other hash functions? Something like:

```rust
let digest = blake3::hash(message);
let mut buf = [0u8; 32];
buf.copy_from_slice(&digest);
buf
```

Reply:
no need.
the internal code in Sha256::digest is 
```
    fn digest(data: impl AsRef<[u8]>) -> Output<Self> {
        let mut hasher = Self::default();
        hasher.update(data.as_ref());
        hasher.finalize()
    }
```
1. new a default hasher
2. update the state using the provided data

in Rust blake3, it don't not offer blake3::digest, reason:
The fact that in cases like this the user has to import some additional traits is indeed annoying
[https://github.com/BLAKE3-team/BLAKE3/issues/11](https://github.com/BLAKE3-team/BLAKE3/issues/11)

so in the BLAKE3/src/test.rs

```
    let digest_str = "04e0bb39f30b1a3feb89f536c93be15055482df748674b00d26e5a75777702e9";
    let mut hasher = crate::Hasher::new();
    hasher.update(b"foo");
    let digest = hasher.finalize();
```